crates/biome_js_analyze/src/analyzers/nursery/use_shorthand_assign.rs

```diff
@@ -0,0 +1,214 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, Rule, RuleDiagnostic,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    AnyJsAssignment, AnyJsExpression, JsAssignmentExpression, JsAssignmentOperator,
+    JsBinaryExpression, JsBinaryOperator, JsSyntaxKind, T,
+};
+use biome_rowan::{AstNode, BatchMutationExt};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Require assignment operator shorthand where possible
+    ///
+    /// JavaScript provides shorthand operators that combine variable assignment and some simple mathematical operations
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/operator-assignment/
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+
+    /// a = a + 1;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+
+    /// a = a - 1;
+    /// ```
+    ///
+    ///  ```js,expect_diagnostic
+
+    /// a = a * 1;
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+
+    /// a += 1;
+    /// ```
+    ///
+    /// ```js
+
+    /// a -= 1;
+    /// ```
+    ///
+    ///  ```js
+
+    /// a *= 1;
+    /// ```
+    ///
+    pub(crate) UseShorthandAssign {
+        version: "next",
+        name: "useShorthandAssign",
+        recommended: false,
+    }
+}
+
+pub struct RuleState {
+    shorthand_operator: JsSyntaxKind,
+    replacement_expression: AnyJsExpression,
+}
+
+#[derive(Clone)]
+enum VariablePosition {
+    Left,
+    Right,
+}
+
+impl Rule for UseShorthandAssign {
+    type Query = Ast<JsAssignmentExpression>;
+    type State = RuleState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        if !matches!(node.operator(), Ok(JsAssignmentOperator::Assign)) {
+            return None;
+        }
+
+        let left = node.left().ok();
+        let left_var_name = match left?.as_any_js_assignment()? {
+            AnyJsAssignment::JsComputedMemberAssignment(assignment) => assignment.text(),
+            AnyJsAssignment::JsIdentifierAssignment(assignment) => assignment.text(),
+            AnyJsAssignment::JsStaticMemberAssignment(assignment) => assignment.text(),
+            _ => return None,
+        };
+
+        let binary_expression = match node.right().ok()? {
```

Comment:
nit: you declare both `left` and `right` at the beginning, so if both emit an error, we exit early without doing any match:

```rs
let left = node.left().ok()?;
let right = node.right().ok()?;
```